server/src/Api/Handlers.hs

```diff
@@ -102,7 +103,14 @@ finalizeTx (FinalizeRequest {tx, datums, redeemers}) = do
   decodedDatums <-
     throwDecodeErrorWithMessage "Failed to decode datums" $
       traverse decodeCborDatum datums
-  let languages = Set.fromList [PlutusV1]
+  let witnessScripts = (Tx.txscripts' . Tx.wits) decodedTx
+      (Value.Value ada maps) = (Tx.mint . Tx.body) decodedTx
+      languages =
+        if Map.null witnessScripts
+          && Map.null maps
+          && ada == 0
```

Comment:
This looks a bit different from a similar change in [another PR](https://github.com/Plutonomicon/cardano-transaction-lib/pull/395/files#diff-4f1478ccd6d4b99631ab8cf039548961315e8091fe4b5dafa1d6561a38bc670eR106).

What documentation this code is based on?

Reply:
@klntsky These things are often undocumented and the changes from the other PR were a related but separate issue